File: dart/2918.dart
class Solution {
    int max(int a, int b){
        return (a > b) ? a : b;
    }

    int minSum(List<int> nums1, List<int> nums2) {
        int s1 = 0, s2 = 0, z1 = 0, z2 = 0;
        for (int i in nums1)
            if (i == 0)
                z1++;
            else
                s1 += i;
        for (int i in nums2)
            if (i == 0)
                z2++;
            else
                s2 += i;
        s1 += z1;
        s2 += z2;
        if (s1 < s2 && z1 == 0)
            return -1;
        if (s1 > s2 && z2 == 0)
            return -1;
        return max(s1, s2);
    }
}

File: java/2918.java
class Solution {
    public long minSum(int[] nums1, int[] nums2) {
        long s1 = 0, s2 = 0;
        int z1 = 0, z2 = 0;
        for (int i : nums1)
            if (i == 0)
                z1++;
            else
                s1 += i;
        for (int i : nums2)
            if (i == 0)
                z2++;
            else
                s2 += i;
        if (s1 < s2) {
            long r = s1 + Math.max(s2 - s1 + z2, z1);
            if ((r != s1 && z1 == 0) || (r > s2 && z2 == 0))
                return -1;
            return r;
        } else {
            long r = s2 + Math.max(s1 - s2 + z1, z2);
            if ((r != s2 && z2 == 0) || (r > s1 && z1 == 0))
                return -1;
            return r;
        }
    }
}
